src/ServiceControl/SagaAudit/AuditRetentionCustomCheck.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace ServiceControl.SagaAudit
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.CustomChecks;
+    using Raven.Client;
+    using ServiceBus.Management.Infrastructure.Settings;
+
+    class AuditRetentionCustomCheck : CustomCheck
+
+    {
+        readonly IDocumentStore _documentStore;
+        readonly bool _auditRetentionPeriodIsSet;
+
+        public AuditRetentionCustomCheck(IDocumentStore documentStore, Settings settings, TimeSpan? repeatAfter = null)
+            : base("Saga audit data retention check", "ServiceControl Health", repeatAfter.HasValue ? repeatAfter : TimeSpan.FromHours(24))
+        {
+            _documentStore = documentStore;
+            _auditRetentionPeriodIsSet = settings.AuditRetentionPeriod.HasValue;
+        }
+
+        public override async Task<CheckResult> PerformCheck()
+        {
+            if (_auditRetentionPeriodIsSet)
+            {
+                return CheckResult.Pass;
+            }
+
+            if (await DetectSagaAuditData().ConfigureAwait(false))
+            {
+                return CheckResult.Failed("Saga snapshot data detected without an audit retention period configured. If saga audit data is allowed to accumulate, it can result in degraded performance.");
```

Comment:
It might be worth linking to docs somewhere (and raising a PR in docs to address this) to explain to the user what to do if they see this message. I think a couple of the other custom checks point to https://docs.particular.net/search?q=servicecontrol+troubleshooting

Reply:
Here is the doc PR. https://github.com/Particular/docs.particular.net/pull/5546